packages/jellyfish-api-core/__tests__/category/blockchain/waitForNewBlock.test.ts

```diff
@@ -0,0 +1,53 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+
+const container = new MasterNodeRegTestContainer()
+const client = new ContainerAdapterClient(container)
+
+describe('new block', () => {
+  beforeAll(async () => {
+    await container.start()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should wait for new block', async () => {
+    {
+      const count = await client.blockchain.getBlockCount()
+      expect(count).toStrictEqual(0)
+    }
+
+    {
+      const promise = client.blockchain.waitForNewBlock()
+      await container.generate(1)
+
+      expect(await promise).toStrictEqual({
+        height: 1,
+        hash: expect.stringMatching(/^[0-f]{64}$/)
+      })
```

Comment:
If I got it correct, this is more useful when some other nodes generate the blocks and the test node is waiting for it to sync.

Reply:
added multi-node testing to both methods